test/e2e/annotations/rewrite.go

```diff
@@ -169,4 +175,106 @@ var _ = framework.IngressNginxDescribe("Annotations - Rewrite", func() {
 		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
 		Expect(body).Should(ContainSubstring(expectBodyRequestURI))
 	})
+
+	It("should use ~* location modifier if regex annotation is present", func() {
+		host := "rewrite.bar.com"
+
+		By("creating a regular ingress definition")
+		ing := framework.NewSingleIngress("kube-lego", "/foo", host, f.IngressController.Namespace, "http-svc", 80, &map[string]string{})
+		_, err := f.EnsureIngress(ing)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ing).NotTo(BeNil())
+
+		err = f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, "/foo")
```

Comment:
this is about location right? maybe be more specific and assert `location /foo` instead?

also, as pointed out above, you can include below assertion here